File: 7 kyu/Count the Digit/main.dart
int nbDig(int n, int d) {
  int res = 0;
  for (int k = 0; k <= n; k++) {
    int temp = k * k;
    res += temp.toString().split('').where((e) => e == d.toString()).length;
  }
  return res;
}

int nbDig(int n, int d) {
  var count = 0;
  for (var i = 0; i <= n; i++) {
    count += d.toString().allMatches((i*i).toString()).length;
  }
  return count;
}

int nbDig(int n, int d) =>
    '$d'.allMatches([for (var i = 0; i <= n; i += 1) i * i].join()).length;

File: 7 kyu/Count the Digit/main.cpp
#include <algorithm>
#include <string>

class CountDig {
public:
    static int nbDig(int n, int d);
};

int CountDig::nbDig(int n, int d)
{
    int count = 0;
    for (int i = 0; i <= n; i++) {
        std::string s = std::to_string(i * i);
        count += std::count(s.begin(), s.end(), std::to_string(d)[0]);
    }
    return count;
}